### PR TITLE
Added x-ms-file-request-itent to PutRangeFromUrl

### PR DESCRIPTION
**Pull request by seanmcc-msft**

### CODE REVIEW

**Review by amnguye — APPROVED** [review]
LGTM - Update the changelog before merging

File: sdk/storage/Azure.Storage.Files.Shares/src/autorest.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ Run `dotnet build /t:GenerateCode` to generate code.
 
 ``` yaml
 input-file:
-    - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/ccae65b66ac23a9fd891ba86f45abc4372a9c688/specification/storage/data-plane/Microsoft.FileStorage/preview/2023-01-03/file.json
+    - https://raw.githubusercontent.com/seanmcc-msft/azure-rest-api-specs/15f80f2475543b1121921bb4c18db1eea3f00055/specification/storage/data-plane/Microsoft.FileStorage/preview/2023-08-03/file.json
```
Comment (seanmcc-msft) [inline sdk/storage/Azure.Storage.Files.Shares/src/autorest.md:7]: TODO update this when swagger PR merges to main.

Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Files.Shares/src/autorest.md:7]: Swagger PR has merged

**Pull request merged**